app/controllers/public_controller.rb

```diff
@@ -8,7 +8,7 @@ def index
   end
   def profile
     community_list
-    
+    @saved_posts = SavePost.where(account_id: current_account.id)
```

Comment:
the SavePost query seems to be duplicated in two places, can you move it to a common place and access it in all the methods instead of creating a duplicate? 
Follow DRY everywhere